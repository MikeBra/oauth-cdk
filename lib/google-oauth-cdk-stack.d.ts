import * as cdk from "aws-cdk-lib";
export declare class GoogleOAuthCdkStack extends cdk.Stack {
    constructor(scope: cdk.App, id: string, props?: cdk.StackProps);
}
