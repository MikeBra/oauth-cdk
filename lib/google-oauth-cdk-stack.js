"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleOAuthCdkStack = void 0;
const cdk = require("aws-cdk-lib");
const lambda = require("aws-cdk-lib/aws-lambda");
const apigateway = require("aws-cdk-lib/aws-apigateway");
const path = require("path");
class GoogleOAuthCdkStack extends cdk.Stack {
    constructor(scope, id, props) {
        super(scope, id, props);
        // Environment variables for the Lambda functions
        const environment = {
            GOOGLE_CLIENT_ID: process.env.GOOGLE_CLIENT_ID || "",
            GOOGLE_CLIENT_SECRET: process.env.GOOGLE_CLIENT_SECRET || "",
            REDIRECT_URI: process.env.REDIRECT_URI || "",
        };
        // OAuth Callback Lambda
        const oauthCallbackLambda = new lambda.Function(this, "OAuthCallbackHandler", {
            runtime: lambda.Runtime.NODEJS_18_X,
            handler: "oauth-callback.handler",
            code: lambda.Code.fromAsset(path.join(__dirname, "../src/lambda")),
            environment,
        });
        // Get User Info Lambda
        const getUserInfoLambda = new lambda.Function(this, "GetUserInfoHandler", {
            runtime: lambda.Runtime.NODEJS_18_X,
            handler: "get-user-info.handler",
            code: lambda.Code.fromAsset(path.join(__dirname, "../src/lambda")),
            environment,
        });
        // API Gateway
        const api = new apigateway.RestApi(this, "GoogleOAuthApi", {
            restApiName: "Google OAuth Service",
            description: "API Gateway for Google OAuth integration",
        });
        // OAuth callback endpoint
        const oauth = api.root.addResource("oauth");
        const callback = oauth.addResource("callback");
        callback.addMethod("GET", new apigateway.LambdaIntegration(oauthCallbackLambda));
        // User info endpoint
        const userInfo = api.root.addResource("userinfo");
        userInfo.addMethod("GET", new apigateway.LambdaIntegration(getUserInfoLambda));
    }
}
exports.GoogleOAuthCdkStack = GoogleOAuthCdkStack;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiZ29vZ2xlLW9hdXRoLWNkay1zdGFjay5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbImdvb2dsZS1vYXV0aC1jZGstc3RhY2sudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7O0FBQUEsbUNBQWtDO0FBQ2xDLGlEQUFnRDtBQUNoRCx5REFBd0Q7QUFDeEQsNkJBQTRCO0FBRTVCLE1BQWEsbUJBQW9CLFNBQVEsR0FBRyxDQUFDLEtBQUs7SUFDakQsWUFBWSxLQUFjLEVBQUUsRUFBVSxFQUFFLEtBQXNCO1FBQzdELEtBQUssQ0FBQyxLQUFLLEVBQUUsRUFBRSxFQUFFLEtBQUssQ0FBQyxDQUFBO1FBRXZCLGlEQUFpRDtRQUNqRCxNQUFNLFdBQVcsR0FBRztZQUNuQixnQkFBZ0IsRUFBRSxPQUFPLENBQUMsR0FBRyxDQUFDLGdCQUFnQixJQUFJLEVBQUU7WUFDcEQsb0JBQW9CLEVBQUUsT0FBTyxDQUFDLEdBQUcsQ0FBQyxvQkFBb0IsSUFBSSxFQUFFO1lBQzVELFlBQVksRUFBRSxPQUFPLENBQUMsR0FBRyxDQUFDLFlBQVksSUFBSSxFQUFFO1NBQzVDLENBQUE7UUFFRCx3QkFBd0I7UUFDeEIsTUFBTSxtQkFBbUIsR0FBRyxJQUFJLE1BQU0sQ0FBQyxRQUFRLENBQzlDLElBQUksRUFDSixzQkFBc0IsRUFDdEI7WUFDQyxPQUFPLEVBQUUsTUFBTSxDQUFDLE9BQU8sQ0FBQyxXQUFXO1lBQ25DLE9BQU8sRUFBRSx3QkFBd0I7WUFDakMsSUFBSSxFQUFFLE1BQU0sQ0FBQyxJQUFJLENBQUMsU0FBUyxDQUFDLElBQUksQ0FBQyxJQUFJLENBQUMsU0FBUyxFQUFFLGVBQWUsQ0FBQyxDQUFDO1lBQ2xFLFdBQVc7U0FDWCxDQUNELENBQUE7UUFFRCx1QkFBdUI7UUFDdkIsTUFBTSxpQkFBaUIsR0FBRyxJQUFJLE1BQU0sQ0FBQyxRQUFRLENBQUMsSUFBSSxFQUFFLG9CQUFvQixFQUFFO1lBQ3pFLE9BQU8sRUFBRSxNQUFNLENBQUMsT0FBTyxDQUFDLFdBQVc7WUFDbkMsT0FBTyxFQUFFLHVCQUF1QjtZQUNoQyxJQUFJLEVBQUUsTUFBTSxDQUFDLElBQUksQ0FBQyxTQUFTLENBQUMsSUFBSSxDQUFDLElBQUksQ0FBQyxTQUFTLEVBQUUsZUFBZSxDQUFDLENBQUM7WUFDbEUsV0FBVztTQUNYLENBQUMsQ0FBQTtRQUVGLGNBQWM7UUFDZCxNQUFNLEdBQUcsR0FBRyxJQUFJLFVBQVUsQ0FBQyxPQUFPLENBQUMsSUFBSSxFQUFFLGdCQUFnQixFQUFFO1lBQzFELFdBQVcsRUFBRSxzQkFBc0I7WUFDbkMsV0FBVyxFQUFFLDBDQUEwQztTQUN2RCxDQUFDLENBQUE7UUFFRiwwQkFBMEI7UUFDMUIsTUFBTSxLQUFLLEdBQUcsR0FBRyxDQUFDLElBQUksQ0FBQyxXQUFXLENBQUMsT0FBTyxDQUFDLENBQUE7UUFDM0MsTUFBTSxRQUFRLEdBQUcsS0FBSyxDQUFDLFdBQVcsQ0FBQyxVQUFVLENBQUMsQ0FBQTtRQUM5QyxRQUFRLENBQUMsU0FBUyxDQUNqQixLQUFLLEVBQ0wsSUFBSSxVQUFVLENBQUMsaUJBQWlCLENBQUMsbUJBQW1CLENBQUMsQ0FDckQsQ0FBQTtRQUVELHFCQUFxQjtRQUNyQixNQUFNLFFBQVEsR0FBRyxHQUFHLENBQUMsSUFBSSxDQUFDLFdBQVcsQ0FBQyxVQUFVLENBQUMsQ0FBQTtRQUNqRCxRQUFRLENBQUMsU0FBUyxDQUNqQixLQUFLLEVBQ0wsSUFBSSxVQUFVLENBQUMsaUJBQWlCLENBQUMsaUJBQWlCLENBQUMsQ0FDbkQsQ0FBQTtJQUNGLENBQUM7Q0FDRDtBQXBERCxrREFvREMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgKiBhcyBjZGsgZnJvbSBcImF3cy1jZGstbGliXCJcbmltcG9ydCAqIGFzIGxhbWJkYSBmcm9tIFwiYXdzLWNkay1saWIvYXdzLWxhbWJkYVwiXG5pbXBvcnQgKiBhcyBhcGlnYXRld2F5IGZyb20gXCJhd3MtY2RrLWxpYi9hd3MtYXBpZ2F0ZXdheVwiXG5pbXBvcnQgKiBhcyBwYXRoIGZyb20gXCJwYXRoXCJcblxuZXhwb3J0IGNsYXNzIEdvb2dsZU9BdXRoQ2RrU3RhY2sgZXh0ZW5kcyBjZGsuU3RhY2sge1xuXHRjb25zdHJ1Y3RvcihzY29wZTogY2RrLkFwcCwgaWQ6IHN0cmluZywgcHJvcHM/OiBjZGsuU3RhY2tQcm9wcykge1xuXHRcdHN1cGVyKHNjb3BlLCBpZCwgcHJvcHMpXG5cblx0XHQvLyBFbnZpcm9ubWVudCB2YXJpYWJsZXMgZm9yIHRoZSBMYW1iZGEgZnVuY3Rpb25zXG5cdFx0Y29uc3QgZW52aXJvbm1lbnQgPSB7XG5cdFx0XHRHT09HTEVfQ0xJRU5UX0lEOiBwcm9jZXNzLmVudi5HT09HTEVfQ0xJRU5UX0lEIHx8IFwiXCIsXG5cdFx0XHRHT09HTEVfQ0xJRU5UX1NFQ1JFVDogcHJvY2Vzcy5lbnYuR09PR0xFX0NMSUVOVF9TRUNSRVQgfHwgXCJcIixcblx0XHRcdFJFRElSRUNUX1VSSTogcHJvY2Vzcy5lbnYuUkVESVJFQ1RfVVJJIHx8IFwiXCIsXG5cdFx0fVxuXG5cdFx0Ly8gT0F1dGggQ2FsbGJhY2sgTGFtYmRhXG5cdFx0Y29uc3Qgb2F1dGhDYWxsYmFja0xhbWJkYSA9IG5ldyBsYW1iZGEuRnVuY3Rpb24oXG5cdFx0XHR0aGlzLFxuXHRcdFx0XCJPQXV0aENhbGxiYWNrSGFuZGxlclwiLFxuXHRcdFx0e1xuXHRcdFx0XHRydW50aW1lOiBsYW1iZGEuUnVudGltZS5OT0RFSlNfMThfWCxcblx0XHRcdFx0aGFuZGxlcjogXCJvYXV0aC1jYWxsYmFjay5oYW5kbGVyXCIsXG5cdFx0XHRcdGNvZGU6IGxhbWJkYS5Db2RlLmZyb21Bc3NldChwYXRoLmpvaW4oX19kaXJuYW1lLCBcIi4uL3NyYy9sYW1iZGFcIikpLFxuXHRcdFx0XHRlbnZpcm9ubWVudCxcblx0XHRcdH1cblx0XHQpXG5cblx0XHQvLyBHZXQgVXNlciBJbmZvIExhbWJkYVxuXHRcdGNvbnN0IGdldFVzZXJJbmZvTGFtYmRhID0gbmV3IGxhbWJkYS5GdW5jdGlvbih0aGlzLCBcIkdldFVzZXJJbmZvSGFuZGxlclwiLCB7XG5cdFx0XHRydW50aW1lOiBsYW1iZGEuUnVudGltZS5OT0RFSlNfMThfWCxcblx0XHRcdGhhbmRsZXI6IFwiZ2V0LXVzZXItaW5mby5oYW5kbGVyXCIsXG5cdFx0XHRjb2RlOiBsYW1iZGEuQ29kZS5mcm9tQXNzZXQocGF0aC5qb2luKF9fZGlybmFtZSwgXCIuLi9zcmMvbGFtYmRhXCIpKSxcblx0XHRcdGVudmlyb25tZW50LFxuXHRcdH0pXG5cblx0XHQvLyBBUEkgR2F0ZXdheVxuXHRcdGNvbnN0IGFwaSA9IG5ldyBhcGlnYXRld2F5LlJlc3RBcGkodGhpcywgXCJHb29nbGVPQXV0aEFwaVwiLCB7XG5cdFx0XHRyZXN0QXBpTmFtZTogXCJHb29nbGUgT0F1dGggU2VydmljZVwiLFxuXHRcdFx0ZGVzY3JpcHRpb246IFwiQVBJIEdhdGV3YXkgZm9yIEdvb2dsZSBPQXV0aCBpbnRlZ3JhdGlvblwiLFxuXHRcdH0pXG5cblx0XHQvLyBPQXV0aCBjYWxsYmFjayBlbmRwb2ludFxuXHRcdGNvbnN0IG9hdXRoID0gYXBpLnJvb3QuYWRkUmVzb3VyY2UoXCJvYXV0aFwiKVxuXHRcdGNvbnN0IGNhbGxiYWNrID0gb2F1dGguYWRkUmVzb3VyY2UoXCJjYWxsYmFja1wiKVxuXHRcdGNhbGxiYWNrLmFkZE1ldGhvZChcblx0XHRcdFwiR0VUXCIsXG5cdFx0XHRuZXcgYXBpZ2F0ZXdheS5MYW1iZGFJbnRlZ3JhdGlvbihvYXV0aENhbGxiYWNrTGFtYmRhKVxuXHRcdClcblxuXHRcdC8vIFVzZXIgaW5mbyBlbmRwb2ludFxuXHRcdGNvbnN0IHVzZXJJbmZvID0gYXBpLnJvb3QuYWRkUmVzb3VyY2UoXCJ1c2VyaW5mb1wiKVxuXHRcdHVzZXJJbmZvLmFkZE1ldGhvZChcblx0XHRcdFwiR0VUXCIsXG5cdFx0XHRuZXcgYXBpZ2F0ZXdheS5MYW1iZGFJbnRlZ3JhdGlvbihnZXRVc2VySW5mb0xhbWJkYSlcblx0XHQpXG5cdH1cbn1cbiJdfQ==