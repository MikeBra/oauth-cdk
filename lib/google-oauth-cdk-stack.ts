import * as cdk from "aws-cdk-lib"
import * as lambda from "aws-cdk-lib/aws-lambda"
import * as apigateway from "aws-cdk-lib/aws-apigateway"
import * as nodejs from "aws-cdk-lib/aws-lambda-nodejs"
import * as path from "path"

export class GoogleOAuthCdkStack extends cdk.Stack {
	constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
		super(scope, id, props)

		// Environment variables for the Lambda functions
		const environment = {
			GOOGLE_CLIENT_ID: process.env.GOOGLE_CLIENT_ID || "",
			GOOGLE_CLIENT_SECRET: process.env.GOOGLE_CLIENT_SECRET || "",
			REDIRECT_URI: process.env.REDIRECT_URI || "",
		}

		// OAuth Callback Lambda
		const oauthCallbackLambda = new lambda.Function(
			this,
			"OAuthCallbackHandler",
			{
				runtime: lambda.Runtime.NODEJS_18_X,
				handler: "oauth-callback.handler",
				code: lambda.Code.fromAsset(path.join(__dirname, "../src/lambda")),
				environment,
			}
		)

		// Get User Info Lambda
		const getUserInfoFunction = new nodejs.NodejsFunction(
			this,
			"GetUserInfoFunction",
			{
				runtime: lambda.Runtime.NODEJS_18_X,
				entry: path.join(__dirname, "../src/lambda/google-get-user-info.ts"),
				handler: "handler",
				environment: {
					GOOGLE_CLIENT_ID: process.env.GOOGLE_CLIENT_ID!,
					GOOGLE_CLIENT_SECRET: process.env.GOOGLE_CLIENT_SECRET!,
				},
			}
		)

		// API Gateway
		const api = new apigateway.RestApi(this, "GoogleOAuthApi", {
			restApiName: "Google OAuth API",
		})

		// OAuth callback endpoint
		const oauth = api.root.addResource("oauth")
		const callback = oauth.addResource("callback")
		callback.addMethod(
			"GET",
			new apigateway.LambdaIntegration(oauthCallbackLambda)
		)

		// User info endpoint
		const userInfo = api.root.addResource("user-info")
		userInfo.addMethod(
			"GET",
			new apigateway.LambdaIntegration(getUserInfoFunction),
			{
				authorizationType: apigateway.AuthorizationType.NONE, // Since we're using Google's tokens
			}
		)

		// Output the API URL
		new cdk.CfnOutput(this, "ApiUrl", {
			value: api.url,
			description: "API Gateway URL",
		})
	}
}
