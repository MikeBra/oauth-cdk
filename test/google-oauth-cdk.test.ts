import * as cdk from "aws-cdk-lib"
import * as GoogleOauthCdk from "../lib/google-oauth-cdk-stack"

// example test. To run these tests, uncomment this file along with the
// example resource in lib/oauth-cdk-stack.ts
test("Empty Stack", () => {
	const app = new cdk.App()
	// WHEN
	const stack = new GoogleOauthCdk.GoogleOAuthCdkStack(app, "MyTestStack")
	// THEN
	const template = app.synth().getStackArtifact(stack.artifactId).template
	expect(template).toMatchSnapshot()
})
