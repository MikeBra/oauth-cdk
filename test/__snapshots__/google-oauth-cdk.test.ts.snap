// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Stack 1`] = `
{
  "Outputs": {
    "GoogleOAuthApiEndpointEA24F466": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "GoogleOAuthApi640967F5",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "GoogleOAuthApiDeploymentStageprod88B87F99",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "GetUserInfoHandler054C8731": {
      "DependsOn": [
        "GetUserInfoHandlerServiceRoleFC1B2691",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "ce192ab3bd7a95563ce770cb772e9ffe8a96af73b567918ff70efecc18cc0b62.zip",
        },
        "Environment": {
          "Variables": {
            "GOOGLE_CLIENT_ID": "test-client-id",
            "GOOGLE_CLIENT_SECRET": "test-client-secret",
            "REDIRECT_URI": "http://localhost:3000/callback",
          },
        },
        "Handler": "get-user-info.handler",
        "Role": {
          "Fn::GetAtt": [
            "GetUserInfoHandlerServiceRoleFC1B2691",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "GetUserInfoHandlerServiceRoleFC1B2691": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "GoogleOAuthApi640967F5": {
      "Properties": {
        "Description": "API Gateway for Google OAuth integration",
        "Name": "Google OAuth Service",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "GoogleOAuthApiAccountF4F2DB13": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "GoogleOAuthApi640967F5",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "GoogleOAuthApiCloudWatchRole1966F53E",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "GoogleOAuthApiCloudWatchRole1966F53E": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "GoogleOAuthApiDeploymentA14C0A5A3ffa8e65a5265a6e42dcb4f319be2f83": {
      "DependsOn": [
        "GoogleOAuthApioauthcallbackGET5441C756",
        "GoogleOAuthApioauthcallback2249B6C8",
        "GoogleOAuthApioauthD9E7E312",
        "GoogleOAuthApiuserinfoGETB7D05841",
        "GoogleOAuthApiuserinfoCCCD2653",
      ],
      "Properties": {
        "Description": "API Gateway for Google OAuth integration",
        "RestApiId": {
          "Ref": "GoogleOAuthApi640967F5",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "GoogleOAuthApiDeploymentStageprod88B87F99": {
      "DependsOn": [
        "GoogleOAuthApiAccountF4F2DB13",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "GoogleOAuthApiDeploymentA14C0A5A3ffa8e65a5265a6e42dcb4f319be2f83",
        },
        "RestApiId": {
          "Ref": "GoogleOAuthApi640967F5",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "GoogleOAuthApioauthD9E7E312": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "GoogleOAuthApi640967F5",
            "RootResourceId",
          ],
        },
        "PathPart": "oauth",
        "RestApiId": {
          "Ref": "GoogleOAuthApi640967F5",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "GoogleOAuthApioauthcallback2249B6C8": {
      "Properties": {
        "ParentId": {
          "Ref": "GoogleOAuthApioauthD9E7E312",
        },
        "PathPart": "callback",
        "RestApiId": {
          "Ref": "GoogleOAuthApi640967F5",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "GoogleOAuthApioauthcallbackGET5441C756": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "OAuthCallbackHandlerB2CE6EDA",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "GoogleOAuthApioauthcallback2249B6C8",
        },
        "RestApiId": {
          "Ref": "GoogleOAuthApi640967F5",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "GoogleOAuthApioauthcallbackGETApiPermissionMyTestStackGoogleOAuthApi89554015GEToauthcallback54DA6F69": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "OAuthCallbackHandlerB2CE6EDA",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GoogleOAuthApi640967F5",
              },
              "/",
              {
                "Ref": "GoogleOAuthApiDeploymentStageprod88B87F99",
              },
              "/GET/oauth/callback",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GoogleOAuthApioauthcallbackGETApiPermissionTestMyTestStackGoogleOAuthApi89554015GEToauthcallback1ED78D52": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "OAuthCallbackHandlerB2CE6EDA",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GoogleOAuthApi640967F5",
              },
              "/test-invoke-stage/GET/oauth/callback",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GoogleOAuthApiuserinfoCCCD2653": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "GoogleOAuthApi640967F5",
            "RootResourceId",
          ],
        },
        "PathPart": "userinfo",
        "RestApiId": {
          "Ref": "GoogleOAuthApi640967F5",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "GoogleOAuthApiuserinfoGETApiPermissionMyTestStackGoogleOAuthApi89554015GETuserinfoE989FE49": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "GetUserInfoHandler054C8731",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GoogleOAuthApi640967F5",
              },
              "/",
              {
                "Ref": "GoogleOAuthApiDeploymentStageprod88B87F99",
              },
              "/GET/userinfo",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GoogleOAuthApiuserinfoGETApiPermissionTestMyTestStackGoogleOAuthApi89554015GETuserinfoFC8D0000": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "GetUserInfoHandler054C8731",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GoogleOAuthApi640967F5",
              },
              "/test-invoke-stage/GET/userinfo",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GoogleOAuthApiuserinfoGETB7D05841": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "GetUserInfoHandler054C8731",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "GoogleOAuthApiuserinfoCCCD2653",
        },
        "RestApiId": {
          "Ref": "GoogleOAuthApi640967F5",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "OAuthCallbackHandlerB2CE6EDA": {
      "DependsOn": [
        "OAuthCallbackHandlerServiceRole9A10E712",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "ce192ab3bd7a95563ce770cb772e9ffe8a96af73b567918ff70efecc18cc0b62.zip",
        },
        "Environment": {
          "Variables": {
            "GOOGLE_CLIENT_ID": "test-client-id",
            "GOOGLE_CLIENT_SECRET": "test-client-secret",
            "REDIRECT_URI": "http://localhost:3000/callback",
          },
        },
        "Handler": "oauth-callback.handler",
        "Role": {
          "Fn::GetAtt": [
            "OAuthCallbackHandlerServiceRole9A10E712",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "OAuthCallbackHandlerServiceRole9A10E712": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
